src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -61,15 +63,20 @@ Status CoreWorkerPlasmaStoreProvider::Create(const std::shared_ptr<Buffer> &meta
   uint32_t delay = RayConfig::instance().object_store_full_initial_delay_ms();
   Status status;
   bool should_retry = true;
+  // If we cannot retry, then always evict on the first attempt.
+  bool evict_if_full = max_retries == 0 ? true : evict_if_full_;
   while (should_retry) {
     should_retry = false;
     arrow::Status plasma_status;
     std::shared_ptr<arrow::Buffer> arrow_buffer;
     {
       std::lock_guard<std::mutex> guard(store_client_mutex_);
-      plasma_status = store_client_.Create(
-          object_id.ToPlasmaId(), data_size, metadata ? metadata->Data() : nullptr,
-          metadata ? metadata->Size() : 0, &arrow_buffer);
+      plasma_status =
+          store_client_.Create(object_id.ToPlasmaId(), evict_if_full, data_size,
+                               metadata ? metadata->Data() : nullptr,
+                               metadata ? metadata->Size() : 0, &arrow_buffer);
+      // Always try to evict after the first attempt.
```

Comment:
This is a little bit scary because it seems plausible to me that the global GC broadcast + python GC + unpin message + unpinning from plasma could take over 1s on a busy cluster. Maybe we should only evict in the last attempt of the exponential backoff?

Reply:
I think evicting after a short timeout is the right decision here since it ensures reasonable performance under moderate memory pressure. If the cluster is under such high memory pressure that the timing matters, I don't think the workload will be stable under any settings.

Reply:
That's a good point but 1s still feels low enough to cause eviction to happen without "real" memory pressure that could.be solved by a gc.collect() due to CPU/network contention delaying messages

Reply:
I think we can resolve that by periodically running gc collect, perhaps every 5 minutes. That would avoid ever hitting 100% memory usage under normal circumstances.

Perhaps we can also choose a middle value like 5 seconds before eviction starts. 30 seconds is much too long.

Reply:
Agreed 30s is definitely too long

Reply:
Maybe we could have a higher timeout for the first time we fall back to LRU (e.g., 5s) and then reduce after the first time LRU is necessary. Seems like a good compromise to avoid the condition I mentioned but still have reasonable perf for workloads with high memory pressure/that work with LRU (e.g., ipython)

Reply:
Hmm I'm okay with extending the LRU to a longer time, but it'd be good to choose the exact number and behavior afterwards based on some real measurements; otherwise it's just guesswork and it will complicate the code unnecessarily.

As a start, how about we just increase `object_store_full_initial_delay_ms` to at least 5s in the startup config if object pinning is disabled?

Reply:
That would effectively mean that every put would take 5s once the object store fills in the ipython use case, right? If that's the case then I'd say let's just leave it at 1s for now.